### PR TITLE
Show OptionSets in interpreter intellisense

### PR DESCRIPTION
**Pull request by lesaltzm**
This pipes the PowerFxConfig object into intellisense, so that it can be used to show suggestions for globals in the interpreter.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Intellisense.cs [hunk 1]
```diff
@@ -22,13 +22,13 @@ namespace Microsoft.PowerFx.Core.Texl.Intellisense
     internal partial class Intellisense : IIntellisense
     {
         protected readonly IReadOnlyList<ISuggestionHandler> _suggestionHandlers;
-        protected readonly EnumStore _enumStore;
+        protected readonly PowerFxConfig _config;
 
-        public Intellisense(EnumStore enumStore, IReadOnlyList<ISuggestionHandler> suggestionHandlers)
+        public Intellisense(PowerFxConfig config, IReadOnlyList<ISuggestionHandler> suggestionHandlers)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Intellisense.cs:27]: >Intellisense(PowerFxConfig config, [](http://example.com/codeflow?start=15&length=34)

Will this be an inssue for PAClient? 
CanvasIntellisense., DataverseIntellisense both impl it.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs [hunk 2]
```diff
@@ -308,6 +311,17 @@ internal virtual void AddCustomSuggestionsForValuePossibilities()
...
         internal virtual void AddCustomSuggestionsForGlobals()
         {
+            foreach (var global in _powerFxConfig.GetSymbols())
+            {
+                var type = DType.Unknown;
+
+                if (global is IExternalOptionSet optionSet)
+                {
+                    type = optionSet.Type;
+                }
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs:321]: Can symbols know their type? 
Seems like every symbol would need that, and having to cast to specific symbol types is brittle.

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs (hunk 2 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs:321]: #287

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Intellisense.cs (hunk 1 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Intellisense.cs]: Yes, updated

**Review by VamsiModem — APPROVED** [review]

**Pull request merged**